main.cpp

```diff
@@ -126,384 +177,498 @@ struct CarWash
+{
+    return powderGrain * bulletCaliber * barrelLength;
+}
+
+int Bullet::impactTarget(int distance, int elevation, int speed, int mass)
+{
+    return speed * mass / (distance * elevation);
+}
+
 struct Stock
 {
-    // Is skeletonized (bool)
     bool isSkeletonized = false;
-    // Is foldable (bool)
     bool isFoldable = true;
-    // Has cheekrest (bool)
     bool hasCheekrest = true;
-    // Has sling point (bool)
     bool hasSlingPoint = true;
-    // Weight (g) (float)
     float weight = 850.48f;
 
-    // Increase control of recoil in gun
-    int increaseControlRecoil(int recoilForce, float gunWeight); // Returns percentage of recoil control increase
-    // Toggle folded/unfolded state
-    void toggleFoldedUnfolded(bool isFoldable); //
-    // Provide support to firer's head when ADS
+    float increaseControlRecoil(int recoilForce, float gunWeight); 
+    void toggleFoldedUnfolded(bool isFoldable);
     void supportHeadWhenADS(bool isAiming = true, float pressureAgainstStock = 0.0f);
 };
-/*
-Thing 10) Assault rifle
-5 properties:
-    1) Barrel
-    2) Scope
-    3) Magazine
-    4) Bullet
-    5) Stock
-3 things it can do: 
-    1) Discharge bullet
-    2) Chamber bullet
-    3) Eject empty magazine
- */
+
+float Stock::increaseControlRecoil(int recoilForce, float gunWeight)
+{
+    return recoilForce + gunWeight;
+}
+
+void Stock::toggleFoldedUnfolded(bool isStockFoldable)
+{
+    bool toggleCommand = true;
+    if (isStockFoldable == true)
+    {
+        if (toggleCommand == true)
+        {
+            std::cout << "Stock folded\n";
+        }
+        else
+        {
+            std::cout << "Fold command cancel\n";
+        }
+    }
+    else
+    {
+        std::cout << "Unable to fold";
+    }
+}
+
+void Stock::supportHeadWhenADS(bool isAiming, float pressureAgainstStock)
+{
+    float supportScore;
+    if (isAiming == true)
+    {
+        supportScore = pressureAgainstStock * pressureAgainstStock;
+    }
+    else
+    {
+        supportScore = 0.0f;
+    }
+}
+
 struct AssaultRifle
 { 
-    // Barrel
     Barrel standardAssault;
 
-    // Scope
     Scope standardSAR;
 
-    // Magazine
     Magazine stanagStandard;
 
-    // Bullet
     Bullet standard556;
 
-    // Stock
     Stock bullpup;
  
-    // Discharge bullet
     void dischargeBullet(bool bulletInChamber, bool isJammed);
-    // Chamber bullet
     void chamberBullet(bool magazineLoaded, std::string bulletType = "FMJ");
-    // Eject empty magazine
     void ejectEmptyMagazine(bool isEjectButtonPressed, bool isMagazineEmpty);
 };
 
+void AssaultRifle::dischargeBullet(bool bulletInChamber, bool isJammed)
+{
+    bool dischargeBullet;
+    if (bulletInChamber == true && isJammed == false)
+    {
+        dischargeBullet = true;
+        std::cout << "Bullet discharged";
+    }
+    else
+    {
+        dischargeBullet = false;
+    }
+    std::cout << dischargeBullet;
+}
```

Comment:
You are always printing out either 0 or 1, regardless of if the bullet was actually discharged, via the last std::cout statement. 
Fix this logic.
Either print 643 and that's it, or print that it was not discharged and don't print anything else.
- [ ] revise